### PR TITLE
scripts: store maintainer in package metadata

### PR DESCRIPTION
**Pull request by aparcar**
The maintainer could be usable for downstream tooling, so start storing
it in the metadata.

Signed-off-by: Paul Spooren <mail@aparcar.org>

### CODE REVIEW

**Comment by champtar** [comment]
Not sure users should contact package maintainer directly, ever. They should open a ticket and cc the maintainer.

**Comment by aparcar** [comment]
I don't plan to promote that and the data is available anyway in [*downstream tools*](https://openwrt.org/packages/pkgdata/htop). I'm just starting to make it package manager agnostic and more parse friendly.

**Comment by champtar** [comment]
Who was maintainer 6 month ago when a package was built is not relevant when opening an issue. Maintainer can have their work email address and change job, maintainer can change or just stop wanting to be maintainer, so we should only look in master branch for maintainer information.
I also don't think maintainer information should be copied outside git, as email can be considered PII and open us to take down requests, and we received such request this year for packages repo.

**Comment by aparcar** [comment]
@champtar I can't follow. The [Packages.manifest](https://downloads.openwrt.org/snapshots/targets/ath79/generic/packages/Packages.manifest) stores this data already. Whoever becomes a maintainer (actually create git commits) must be willing to provide some sort of mail address.

Whoever takes over maintainership of a package should update the address in all other active branches as well (whenever backporting).

If it's a PII problem, which seems wild to me, we should tackle this differently. Either remove `Maintainer` from Packages.manifest or ask for company approval to have the mail address used and if not, require the maintainers personal email address.

**Comment by champtar** [comment]
> @champtar I can't follow. The [Packages.manifest](https://downloads.openwrt.org/snapshots/targets/ath79/generic/packages/Packages.manifest) stores this data already. Whoever becomes a maintainer (actually create git commits) must be willing to provide some sort of mail address.

Maybe it should not :)
If a maintainer changes, old maintainer email address will be on GL.inet (random company shipping OpenWrt) repo for months/years, it doesn't make sense.

> Whoever takes over maintainership of a package should update the address in all other active branches as well (whenever backporting).

Still old artifacts keep old (useless) infos

> If it's a PII problem, which seems wild to me, we should tackle this differently. Either remove `Maintainer` from Packages.manifest or ask for company approval to have the mail address used and if not, require the maintainers personal email address.

If I recall correctly the claim that we received was to take down some patches because they contained a specific email in the SoB, claim was sent by the author of said patches (no usurpation of identity), patches that were also sent on some kernel mailing list, luckily the person withdraw after some day, so yes this can be pretty wild.

I don't want to introduce a CLA, there is no need to know the maintainer outside of opening an issue and merging PR so I fail to see why we need the info outside of git. Any takedown against our git repo we can show that they requested to add the email themselves, but we can't claim that maintainer want their email in luci at some point.

Instead of having Maintainer we should add the url to the correct bug tracker and promote this info (maybe using some redirection like openwrt.org/bug-packages to be future proof)

**Comment by aparcar** [comment]
I think the issue you're describing is unrelated. If somebody fakes to be an identity they aren't, people may complain. On the other hand, if you (as a maintainer) consciously add your email address (and copyright) to a public repository, thinks will be public. People use and fork the code, including your name and email,  so it ends up all around the Internet. If you don't want that, use email addresses you don't use outside your open source contributions or ask people of mutual trust to publish your patches.

The point of having a maintainer assigned is the persons reach-ability. If we keep it *in Git* we make it harder for regular users. Looking at AlpineLinux, they have a [very nice setup](https://pkgs.alpinelinux.org/flag/main/perl-convert-asn1/0.33-r0) to report issues with packages without people needing to understand Git nor having an account on GitHub.

If you don't want to be *annoyed* by people having trouble with your maintained package, don't become the maintainer but contribute patches and test things to help existing maintainers.

tl;dr all this data is online already, this patch doesn't change that. If you don't want to receive messages don't do something that requires people to be in touch with you (e.g. being a maintainer)

**Comment by champtar** [comment]
To be clear, will the maintainer info be copied in the ipk or in another file (and which one)

**Comment by aparcar** [comment]
All of this is already the case:

> will the maintainer info be copied in the ipk

```
cat /usr/lib/opkg/info/odhcpd-ipv6only.control | grep Maintainer
Maintainer: Hans Dedecker <dedeckeh@gmail.com>
```
>  ...in another file (and which one)

Currently you find maintainers in the following other files (I know of):
* [All Packages.manifest files](https://downloads.openwrt.org/snapshots/targets/ath79/generic/packages/Packages.manifest)
* [Official packages table](https://openwrt.org/packages/table/start)
* [Official packages csv](https://openwrt.org/_media/packages_dump_tab_separated.zip)
* [External package trackers](https://repology.org/project/odhcpd/versions)
* Surely more at other places

**Comment by champtar** [comment]
If the genie is already out of the bottle please go ahead with this PR, I'll start a discussion on the mailing list
FWIW opkg info doesn't display this info
```
# opkg info odhcpd-ipv6only
Package: odhcpd-ipv6only
Version: 2021-07-18-bc9d317f-3
Depends: libc, libubox20210516, libuci20130104, libubus20210630, libnl-tiny1
Status: install user installed
Architecture: arm_cortex-a9_vfpv3-d16
Installed-Time: 1630265432
```

**Comment by aparcar** [comment]
@champtar please add me as CC for this discussion.
I'm not sure how opkg handles the `info` command, however grepping through the code I found some handling of `Maintainer`. So maybe there is an extra option to show it... In any case, it's stored locally.
